### PR TITLE
refine EngineIOConverter, and use io_convert in test_trt_activation_op

### PR DESCRIPTION
**Pull request by luotao1**
1. change `EngineInputConverter` to `EngineIOConverter`, which has two function: `ConvertInput`(LoDTensor->ITensor) and `ConvertOutput`(ITensor -> LoDTensor)
2. improve the unit-test `test_io_converter.cc`
3. use `EngineIOConverter` in unit-test `test_activation_op.cc`
4. remove duplicated `cudaMemcpyAsync` in `SetInputFromCPU`

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/convert/io_converter.cc [hunk 1]
```diff
@@ -23,34 +23,50 @@ namespace tensorrt {
...
-
+      PADDLE_ENFORCE_EQ(0, cudaMemcpyAsync(out, in.data<float>(), size,
+                                           cudaMemcpyHostToDevice, *stream_));
     } else if (is_gpu_place(place)) {
-      PADDLE_ENFORCE_EQ(0,
-                        cudaMemcpyAsync(out, in.data<float>(), in.memory_size(),
-                                        cudaMemcpyHostToHost, *stream_));
-
+      PADDLE_ENFORCE_EQ(0, cudaMemcpyAsync(out, in.data<float>(), size,
+                                           cudaMemcpyHostToHost, *stream_));
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/io_converter.cc]: not DeiveceToDevice?

File: paddle/fluid/inference/tensorrt/convert/io_converter.h [hunk 2]
```diff
@@ -25,43 +26,57 @@ namespace tensorrt {
...
- * most cases just need to copy the data.
+ * Convert Input from Fluid to TensorRT Engine.
+ * Convert Output from TensorRT Engine to Fluid.
+ *
+ * Note that TensorRT's ITensor follows row major, NCHW. Fluid is also row
+ * major,
+ * so in the default case just need to copy the data.
  */
-class EngineInputConverter {
+class EngineIOConverter {
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/io_converter.h]: Add comments about how to register a convter from fluid to TRT or from TRT to fluid.

For example, a converter for a special Op called `x` that from fluid to TRT will register as `Fluid(x)->TRT`.

Or a special Layer in TRT called `y` that convert output from TRT to fluid will register as `TRT(y)->Fluid`.

File: paddle/fluid/inference/tensorrt/convert/test_activation_op.cc [hunk 3]
```diff
@@ -26,7 +27,7 @@ namespace paddle {
 namespace inference {
 namespace tensorrt {
 
-void Compare(float input, float expect) {
+void Compare(const std::string op_type, float input, float expect) {
```
Comment (Superjomn) [inline paddle/fluid/inference/tensorrt/convert/test_activation_op.cc]: const string&

Comment (luotao1) [reply paddle/fluid/inference/tensorrt/convert/test_activation_op.cc]: I will correct in next PR.

**Pull request merged**